events/filters.py

```diff
@@ -24,3 +28,7 @@ def filter_by_upcoming(self, qs, name, value):
         if value:
             return qs.filter(**{name + "__gte": timezone.now()})
         return qs
+
+    def filter_by_venue_global_id(self, qs, name, value):
+        venue_id = from_global_id(value)[1]
+        return qs.filter(venue=venue_id)
```

Comment:
would `filter(venue_id=venue_id) be more reliable? since you are not passing the `Venue` instance itself.